src/assets.rs

```diff
@@ -0,0 +1,364 @@
+    /// Check if the asset is verified.
+    pub fn verified(&self) -> bool {
+        self.verified
+    }
+
+    /// Check if the asset is temporary.
+    pub fn temporary(&self) -> bool {
+        self.temporary
+    }
+}
+
+type AssetsStore = KeyValueStore<AssetCode, Asset>;
+
+/// An editor to create or update the asset or assets store.
+pub struct AssetEditor<'a> {
+    asset: Asset,
+    store: &'a mut AssetsStore,
+}
+
+impl<'a> AssetEditor<'a> {
+    /// Create an asset editor.
+    fn new(store: &'a mut AssetsStore, asset: Asset) -> Self {
+        Self { asset, store }
+    }
+
+    /// Set the optional asset name.
+    pub fn set_name(mut self, name: Option<String>) -> Self {
+        self.asset.name = name;
+        self
+    }
+
+    /// Set the asset name.
+    pub fn with_name(mut self, name: String) -> Self {
+        self.asset.name = Some(name);
+        self
+    }
+
+    /// Clear the asset name.
+    pub fn clear_name(mut self) -> Self {
+        self.asset.name = None;
+        self
+    }
+
+    /// Set the optional asset description.
+    pub fn set_description(mut self, description: Option<String>) -> Self {
+        self.asset.description = description;
+        self
+    }
+
+    /// Set the asset description.
+    pub fn with_description(mut self, description: String) -> Self {
+        self.asset.description = Some(description);
+        self
+    }
+
+    /// Clear the asset description.
+    pub fn clear_description(mut self) -> Self {
+        self.asset.description = None;
+        self
+    }
+
+    /// Set the optional asset icon.
+    pub fn set_icon(mut self, icon: Option<Icon>) -> Self {
+        self.asset.icon = icon;
+        self
+    }
+
+    /// Set the asset icon.
+    pub fn with_icon(mut self, icon: Icon) -> Self {
+        self.asset.icon = Some(icon);
+        self
+    }
+
+    /// Clear the asset icon.
+    pub fn clear_icon(mut self) -> Self {
+        self.asset.icon = None;
+        self
+    }
+
+    /// Save the asset to the store if it's nontemporary.
+    ///
+    /// If the store has an asset with the same code, merges with it.
+    ///
+    /// Returns the stored asset.
+    pub fn save<L: Ledger>(&mut self) -> Result<Asset, KeystoreError<L>> {
+        self.update();
+        if !self.asset.temporary {
+            self.store.store(&self.asset.definition.code, &self.asset)?;
+        }
+        Ok(self.asset.clone())
+    }
+
+    /// If there exists an asset in the store with the same code as the editor asset, updates the
+    /// editor asset by merging in the store asset.
+    /// * Keeps the asset name, description or icon of the editor asset if
+    ///   * the editor asset is verified and the store asset is not, and
+    ///   * the corresponding attribute exists.
+    /// * Keeps the mint information of the editor asset, if present.
+    /// * Sets as verified if either asset if verified.
+    /// * Sets as temporary if both assets are temporary.
+    pub fn update(&mut self) {
```

Comment:
I think this is a bit more convenient as a function to update with data from a given asset, rather than from the asset in the store. As in `pub fn update(&mut self, other: Asset)`. Then the caller can always load `store_asset` from the store and call `editor.update(store_asset)` if they want (in particular, `create` will do this), but it can also be used if you already have an `Asset` without loading it from the store.